cmd/thanos/main.go

```diff
@@ -115,6 +116,16 @@ func main() {
 		logger = log.With(logger, "ts", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
 	}
 
+	loggerAdapter := func(template string, args ...interface{}) {
+		level.Debug(logger).Log("msg", fmt.Sprintf(template, args))
+	}
+
+	undo, err := maxprocs.Set(maxprocs.Logger(loggerAdapter))
```

Comment:
Can we comment on what are rationales? Why we are doing this? 

There is almost 0 docs on https://github.com/uber-go/automaxprocs

Reply:
@2nick let's add more comments here, pls

Reply:
@bwplotka yep, you're right, app running in container will see all CPUs on host. `maxprocs.Set()` calculates GOMAXPROCS more accurate because also uses cgroup limits.
It's pretty convenient to configure CPU limits in one place when you're setting them for container and don't mind about GOMAXPROCS at all :)

Reply:
Added comment and force-pushed amended commit - will github switch to the new commit automatically or I need to make some additional actions? First time merge requesting so sorry for newbie questions. :)

Reply:
It's ok, GitHub will take care of this automatically :+1: